nncf/common/tensor_statistics/collectors.py

```diff
@@ -508,39 +545,52 @@ def __init__(
         self._use_means_of_mins = use_means_of_mins
         self._use_means_of_maxs = use_means_of_maxs
 
-    def _min_aggregate(self):
-        stacked_min = self._tensor_processor.stack(self._all_min_values)
-        if self._use_means_of_mins:
+    def _min_aggregate(self) -> None:
+        if self._tensor_processor is not None:
+            stacked_min = self._tensor_processor.stack(self._all_min_values)
+        if self._use_means_of_mins and self._tensor_processor is not None:
             return self._tensor_processor.mean(stacked_min, axis=0)
-        return self._tensor_processor.reduce_min(stacked_min, axis=0)
+        if self._tensor_processor is not None:
+            return self._tensor_processor.reduce_min(stacked_min, axis=0)
 
-    def _max_aggregate(self):
-        stacked_max = self._tensor_processor.stack(self._all_max_values)
-        if self._use_means_of_maxs:
+    def _max_aggregate(self) -> None:
+        if self._tensor_processor is not None:
+            stacked_max = self._tensor_processor.stack(self._all_max_values)
+        if self._use_means_of_maxs and self._tensor_processor is not None:
             return self._tensor_processor.mean(stacked_max, axis=0)
-        return self._tensor_processor.reduce_max(stacked_max, axis=0)
+        if self._tensor_processor is not None:
+            return self._tensor_processor.reduce_max(stacked_max, axis=0)
 
 
 class MeanMinMaxStatisticCollector(MinMaxOfflineStatisticCollectorBase):
     """
     Collector aggregates mean of minimum values and mean of maximum values.
     """
 
-    def _min_aggregate(self):
-        stacked_min = self._tensor_processor.stack(self._all_min_values)
-        return self._tensor_processor.mean(stacked_min, axis=0)
+    def _min_aggregate(self) -> None:
+        if self._tensor_processor is not None:
+            stacked_min = self._tensor_processor.stack(self._all_min_values)
+        if self._tensor_processor is not None:
+            return self._tensor_processor.mean(stacked_min, axis=0)
```

Comment:
I think this `if self._tensor_processor is not None:` should be done in a different place